### PR TITLE
refine some confignetwork test cases to avoid using the same bogus IP address

### PR DESCRIPTION
**Pull request by tingtli**
### The PR is to fix task https://github.com/xcat2/xcat2-task-management/issues/591
The pr updated following cases
```
confignetwork_s_installnic_secondarynic_updatenode
confignetwork_secondarynic_nicextraparams_updatenode
confignetwork_2eth_bridge_br0
confignetwork_2eth_bridge_br22_br33
confignetwork_bond_eth2_eth3
confignetwork_bond_false
confignetwork_vlan_bond
confignetwork_vlan_eth0
```
The UT result

[xcattest.log.20190227025704.log](https://github.com/xcat2/xcat-core/files/2909139/xcattest.log.20190227025704.log)

### CODE REVIEW

File: xCAT-test/autotest/testcase/confignetwork/cases0 [hunk 1]
```diff
@@ -160,20 +160,24 @@ cmd:xdsh $$CN "mkdir -p /tmp/backupnet/"
...
 check:rc==0
-cmd:mkdef -t network -o 11_1_0_0-255_255_0_0 net=11.1.0.0 mask=255.255.0.0 mgtifname=$$SECONDNIC
+#cmd:mkdef -t network -o 11_1_0_0-255_255_0_0 net=11.1.0.0 mask=255.255.0.0 mgtifname=$$SECONDNIC
+cmd:cnip=__GETNODEATTR($$CN,ip)__;echo $cnip;var3=100.;var4=`echo $cnip |awk -F. '{print $2}'`;var5=.0.0;secondnet=$var3$var4$var5;mkdef -t network -o confignetworks_test1 net=$secondnet mask=255.255.0.0 mgtifname=$$SECONDNIC
 check:rc==0
-cmd:chdef $$CN nicips.$$SECONDNIC=11.1.0.100 nictypes.$$SECONDNIC=Ethernet nicnetworks.$$SECONDNIC=11_1_0_0-255_255_0_0
+#cmd:chdef $$CN nicips.$$SECONDNIC=11.1.0.100 nictypes.$$SECONDNIC=Ethernet nicnetworks.$$SECONDNIC=11_1_0_0-255_255_0_0
+cmd:cnip=__GETNODEATTR($$CN,ip)__;echo $cnip;var1=100;var2=`echo $cnip |sed -r 's/[^\.]*(..\..*)/\1/'`;second1ip=$var1$var2;chdef $$CN nicips.$$SECONDNIC=$second1ip nictypes.$$SECONDNIC=Ethernet nicnetworks.$$SECONDNIC=confignetworks_test1
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/confignetwork/cases0]: There is a easier way to calculate ``second1ip``

```
cnip=__GETNODEATTR($$CN,ip)__;echo $cnip;econd1ip="100."${cnip#*\.};
```

File: xCAT-test/autotest/testcase/confignetwork/cases0 [hunk 2]
```diff
@@ -160,20 +160,24 @@ cmd:xdsh $$CN "mkdir -p /tmp/backupnet/"
 check:rc==0
 cmd:if grep SUSE /etc/*release;then xdsh $$CN "cp -f /etc/sysconfig/network/ifcfg-* /tmp/backupnet/"; elif grep -E "Red Hat|CentOS" /etc/*release;then xdsh $$CN "cp -f /etc/sysconfig/network-scripts/ifcfg-* /tmp/backupnet/"; elif grep Ubuntu /etc/*release;then xdsh $$CN "cp -f /etc/network/interfaces.d/* /tmp/backupnet/;cp -f /etc/network/interfaces /tmp";else echo "Sorry,this is not supported os"; fi
 check:rc==0
-cmd:mkdef -t network -o 11_1_0_0-255_255_0_0 net=11.1.0.0 mask=255.255.0.0 mgtifname=$$SECONDNIC
+#cmd:mkdef -t network -o 11_1_0_0-255_255_0_0 net=11.1.0.0 mask=255.255.0.0 mgtifname=$$SECONDNIC
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/confignetwork/cases0]: It is better to remove the lines which has been commented out.

File: xCAT-test/autotest/testcase/confignetwork/cases0 [hunk 3]
```diff
@@ -160,20 +160,24 @@ cmd:xdsh $$CN "mkdir -p /tmp/backupnet/"
...
 check:rc==0
 cmd:updatenode $$CN -P "confignetwork -s"
 check:rc==0
-cmd:if grep SUSE /etc/*release;then xdsh $$CN "grep 11.1.0.100 /etc/sysconfig/network/ifcfg-$$SECONDNIC"; elif grep -E "Red Hat|CentOS" /etc/*release;then xdsh $$CN  "grep 11.1.0.100 /etc/sysconfig/network-scripts/ifcfg-$$SECONDNIC"; elif grep Ubuntu /etc/*release;then xdsh $$CN "grep 11.1.0.100 /etc/network/interfaces.d/$$SECONDNIC";else echo "Sorry,this is not supported os"; fi
-check:output=~11.1.0.100
+cmd:cnip=__GETNODEATTR($$CN,ip)__;echo $cnip;var1=100;var2=`echo $cnip |sed -r 's/[^\.]*(..\..*)/\1/'`;second1ip=$var1$var2;if grep SUSE /etc/*release;then xdsh $$CN "grep $second1ip /etc/sysconfig/network/ifcfg-$$SECONDNIC"; elif grep -E "Red Hat|CentOS" /etc/*release;then xdsh $$CN  "grep $second1ip /etc/sysconfig/network-scripts/ifcfg-$$SECONDNIC"; elif grep Ubuntu /etc/*release;then xdsh $$CN "grep $second1ip /etc/network/interfaces.d/$$SECONDNIC";else echo "Sorry,this is not supported os"; fi
+check:rc==0
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/confignetwork/cases0]: If enter branch ``echo "Sorry,this is not supported os";`` the return code still is 0;
It is better to add ``exit 1`` in ``else echo "Sorry,this is not supported os";`` branch

Comment (tingtli) [reply xCAT-test/autotest/testcase/confignetwork/cases0]: if this is not supported os, that means this case does not support the scenario, but not this case fails. So, i don't think we should modify the return code.

File: xCAT-test/autotest/testcase/confignetwork/cases0 [hunk 4]
```diff
@@ -160,20 +160,20 @@ cmd:xdsh $$CN "mkdir -p /tmp/backupnet/"
 check:rc==0
 cmd:if grep SUSE /etc/*release;then xdsh $$CN "cp -f /etc/sysconfig/network/ifcfg-* /tmp/backupnet/"; elif grep -E "Red Hat|CentOS" /etc/*release;then xdsh $$CN "cp -f /etc/sysconfig/network-scripts/ifcfg-* /tmp/backupnet/"; elif grep Ubuntu /etc/*release;then xdsh $$CN "cp -f /etc/network/interfaces.d/* /tmp/backupnet/;cp -f /etc/network/interfaces /tmp";else echo "Sorry,this is not supported os"; fi
 check:rc==0
-cmd:mkdef -t network -o 11_1_0_0-255_255_0_0 net=11.1.0.0 mask=255.255.0.0 mgtifname=$$SECONDNIC
+cmd:cnip=__GETNODEATTR($$CN,ip)__;echo $cnip;var3=100.;var4=`echo $cnip |awk -F. '{print $2}'`;var5=.0.0;secondnet=$var3$var4$var5;mkdef -t network -o confignetworks_test1 net=$secondnet mask=255.255.0.0 mgtifname=$$SECONDNIC
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/confignetwork/cases0]: Depending on our ip naming rule, the algorithm of ``secondnet=$var3$var4$var5`` can not generate unique ip.

File: xCAT-test/autotest/testcase/confignetwork/cases0 [hunk 5]
```diff
@@ -266,11 +266,11 @@ check:rc==0
 cmd:if grep SUSE /etc/*release;then xdsh $$CN "cp -f /etc/sysconfig/network/ifcfg-* /tmp/backupnet/"; elif grep -E "Red Hat|CentOS" /etc/*release;then xdsh $$CN "cp -f /etc/sysconfig/network-scripts/ifcfg-* /tmp/backupnet/"; elif grep Ubuntu /etc/*release;then xdsh $$CN "cp -f /etc/network/interfaces.d/* /tmp/backupnet/";else echo "Sorry,this is not supported os"; fi
 check:rc==0
 cmd:cp -f /etc/hosts /etc/hosts.bak
-cmd:mkdef -t network -o 11_1_0_0-255_255_0_0 net=11.1.0.0 mask=255.255.0.0 mgtifname=$$SECONDNIC
+cmd:cnip=__GETNODEATTR($$CN,ip)__;echo $cnip;var3=100.;var4=`echo $cnip |awk -F. '{print $2}'`;var5=.0.0;secondnet=$var3$var4$var5;mkdef -t network -o confignetworks_test1 net=$secondnet mask=255.255.0.0 mgtifname=$$SECONDNIC
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/confignetwork/cases0]: Depending on our ip naming rule, the algorithm of ``secondnet=$var3$var4$var5`` can not generate unique ip.

File: xCAT-test/autotest/testcase/confignetwork/cases0 [hunk 6]
```diff
@@ -266,11 +266,11 @@ check:rc==0
...
 check:rc==0
-cmd:mkdef -t network -o 12_1_0_0-255_255_0_0 net=12.1.0.0 mask=255.255.0.0 mgtifname=$$SECONDNIC
+cmd:cnip=__GETNODEATTR($$CN,ip)__;echo $cnip;var3=101.;var4=`echo $cnip |awk -F. '{print $2}'`;var5=.0.0;secondnet=$var3$var4$var5;mkdef -t network -o confignetworks_test2 net=$secondnet mask=255.255.0.0 mgtifname=$$SECONDNIC
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/confignetwork/cases0]: Depending on our ip naming rule, the algorithm of ``secondnet=$var3$var4$var5`` can not generate unique ip.

File: xCAT-test/autotest/testcase/confignetwork/cases0 [hunk 7]
```diff
@@ -491,38 +494,38 @@ cmd:xdsh $$CN "mkdir -p /tmp/backupnet/"
 check:rc==0
 cmd:if grep SUSE /etc/*release;then xdsh $$CN "cp -f /etc/sysconfig/network/ifcfg-* /tmp/backupnet/"; elif grep -E "Red Hat|CentOS" /etc/*release;then xdsh $$CN "cp -f /etc/sysconfig/network-scripts/ifcfg-* /tmp/backupnet/"; elif grep Ubuntu /etc/*release;then xdsh $$CN "cp -f /etc/network/interfaces.d/* /tmp/backupnet/";else echo "Sorry,this is not supported os"; fi
 cmd:if grep -E "Red Hat|CentOS" /etc/*release;then xdsh $$CN "yum -y install bridge-utils";fi
-cmd:mkdef -t network -o 11_1_0_0-255_255_0_0 net=11.1.0.0 mask=255.255.0.0 mgtifname=$$SECONDNIC
+cmd:cnip=__GETNODEATTR($$CN,ip)__;echo $cnip;var3=100.;var4=`echo $cnip |awk -F. '{print $2}'`;var5=.0.0;secondnet=$var3$var4$var5;mkdef -t network -o confignetworks_test1 net=$secondnet mask=255.255.0.0 mgtifname=$$SECONDNIC
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/confignetwork/cases0]: Depending on our ip naming rule, the algorithm of secondnet=$var3$var4$var5 can not generate unique ip.

File: xCAT-test/autotest/testcase/confignetwork/cases0 [hunk 8]
```diff
@@ -491,38 +494,38 @@ cmd:xdsh $$CN "mkdir -p /tmp/backupnet/"
...
-check:output=~11.1.0.100
+cmd:cnip=__GETNODEATTR($$CN,ip)__;echo $cnip;var1=100;var2=`echo $cnip |sed -r 's/[^\.]*(..\..*)/\1/'`;second1ip=$var1$var2;if grep SUSE /etc/*release;then xdsh $$CN "grep $second1ip /etc/sysconfig/network/ifcfg-$$SECONDNIC"; elif grep -E "Red Hat|CentOS" /etc/*release;then xdsh $$CN  "grep $second1ip /etc/sysconfig/network-scripts/ifcfg-$$SECONDNIC"; elif grep Ubuntu /etc/*release;then xdsh $$CN "grep $second1ip /etc/network/interfaces.d/$$SECONDNIC";else echo "Sorry,this is not supported os"; fi
+check:rc==0
 cmd:chdef $$CN nicips.$$SECONDNIC= nictypes.$$SECONDNIC= nicnetworks.$$SECONDNIC=
 check:rc==0
-cmd:mkdef -t network -o 60_0_0_0-255_0_0_0 net=60.0.0.0 mask=255.0.0.0 mgtifname=$$SECONDNIC
+cmd:cnip=__GETNODEATTR($$CN,ip)__;echo $cnip;var3=105.;var4=`echo $cnip |awk -F. '{print $2}'`;var5=.0.0;secondnet=$var3$var4$var5;mkdef -t network -o confignetworks_test6 net=$secondnet mask=255.255.0.0 mgtifname=$$SECONDNIC
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/confignetwork/cases0]: Depending on our ip naming rule, the algorithm of ``secondnet=$var3$var4$var5`` can not generate unique ip.

File: xCAT-test/autotest/testcase/confignetwork/cases0 [hunk 9]
```diff
@@ -491,38 +494,38 @@ cmd:xdsh $$CN "mkdir -p /tmp/backupnet/"
...
 check:rc==0
-cmd:mkdef -t network -o 70_0_0_0-255_0_0_0 net=70.0.0.0 mask=255.0.0.0 mgtifname=$$SECONDNIC
+cmd:cnip=__GETNODEATTR($$CN,ip)__;echo $cnip;var3=106.;var4=`echo $cnip |awk -F. '{print $2}'`;var5=.0.0;secondnet=$var3$var4$var5;mkdef -t network -o confignetworks_test7 net=$secondnet mask=255.255.0.0 mgtifname=$$SECONDNIC
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/confignetwork/cases0]: Depending on our ip naming rule, the algorithm of ``secondnet=$var3$var4$var5`` can not generate unique ip.

File: xCAT-test/autotest/testcase/confignetwork/cases0 [hunk 10]
```diff
@@ -588,39 +591,38 @@ cmd:xdsh $$CN "mkdir -p /tmp/backupnet/"
 check:rc==0
 cmd:if grep SUSE /etc/*release;then xdsh $$CN "cp -f /etc/sysconfig/network/ifcfg-* /tmp/backupnet/"; elif grep -E "Red Hat|CentOS" /etc/*release;then xdsh $$CN "cp -f /etc/sysconfig/network-scripts/ifcfg-* /tmp/backupnet/"; elif grep Ubuntu /etc/*release;then xdsh $$CN "cp -f /etc/network/interfaces.d/* /tmp/backupnet/";else echo "Sorry,this is not supported os"; fi
 cmd:if grep -E "Red Hat|CentOS" /etc/*release;then xdsh $$CN "yum -y install bridge-utils";fi
-cmd:mkdef -t network -o 11_1_0_0-255_255_0_0 net=11.1.0.0 mask=255.255.0.0 mgtifname=$$SECONDNIC
+cmd:cnip=__GETNODEATTR($$CN,ip)__;echo $cnip;var3=100.;var4=`echo $cnip |awk -F. '{print $2}'`;var5=.0.0;secondnet=$var3$var4$var5;mkdef -t network -o confignetworks_test1 net=$secondnet mask=255.255.0.0 mgtifname=$$SECONDNIC
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/confignetwork/cases0]: Depending on our ip naming rule, the algorithm of ``secondnet=$var3$var4$var5`` can not generate unique ip.

File: xCAT-test/autotest/testcase/confignetwork/cases0 [hunk 11]
```diff
@@ -588,39 +591,38 @@ cmd:xdsh $$CN "mkdir -p /tmp/backupnet/"
...
 check:rc==0
-cmd:chdef $$CN nicips.$$SECONDNIC=11.1.0.100 nictypes.$$SECONDNIC=Ethernet nicnetworks.$$SECONDNIC=11_1_0_0-255_255_0_0
+cmd:cnip=__GETNODEATTR($$CN,ip)__;echo $cnip;var1=100;var2=`echo $cnip |sed -r 's/[^\.]*(..\..*)/\1/'`;second1ip=$var1$var2;chdef $$CN nicips.$$SECONDNIC=$second1ip nictypes.$$SECONDNIC=Ethernet nicnetworks.$$SECONDNIC=confignetworks_test1
 check:rc==0
-cmd:mkdef -t network -o 12_1_0_0-255_255_0_0 net=12.1.0.0 mask=255.255.0.0 mgtifname=$$THIRDNIC
+cmd:cnip=__GETNODEATTR($$CN,ip)__;echo $cnip;var3=101.;var4=`echo $cnip |awk -F. '{print $2}'`;var5=.0.0;secondnet=$var3$var4$var5;mkdef -t network -o confignetworks_test2 net=$secondnet mask=255.255.0.0 mgtifname=$$THIRDNIC
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/confignetwork/cases0]: Depending on our ip naming rule, the algorithm of ``secondnet=$var3$var4$var5`` can not generate unique ip.

**Comment by hu-weihua** [comment]
It is ok for me, thanks @tingtli

**Pull request merged**